### PR TITLE
treetab: tiny cleanup

### PR DESCRIPTION
**Pull request by Clyybber**

### CODE REVIEW

**Comment by Araq** [comment]
Er, keep in mind that in old Nim this is an expensive copy and `swap` isn't.

File: compiler/treetab.nim [hunk 1]
```diff
@@ -86,12 +86,11 @@ proc nodeTablePut*(t: var TNodeTable, key: PNode, val: int) =
...
   else:
     if mustRehash(t.data.len, t.counter):
-      var n: TNodePairSeq
-      newSeq(n, t.data.len * GrowthFactor)
+      var n = newSeq[TNodePairSeq](t.data.len * GrowthFactor)
       for i in 0..high(t.data):
         if t.data[i].key != nil:
           nodeTableRawInsert(n, t.data[i].h, t.data[i].key, t.data[i].val)
-      swap(t.data, n)
+      t.data = n
```
Comment (Araq) [inline compiler/treetab.nim]:
```suggestion
      t.data = move n
```

File: compiler/treetab.nim [hunk 2]
```diff
@@ -103,12 +102,11 @@ proc nodeTableTestOrSet*(t: var TNodeTable, key: PNode, val: int): int =
...
```
Comment (Araq) [inline compiler/treetab.nim]:
```suggestion
      t.data = move n
```

File: compiler/treetab.nim [hunk 3]
```diff
@@ -86,12 +86,11 @@ proc nodeTablePut*(t: var TNodeTable, key: PNode, val: int) =
     t.data[index].val = val
   else:
     if mustRehash(t.data.len, t.counter):
-      var n: TNodePairSeq
-      newSeq(n, t.data.len * GrowthFactor)
+      var n = newSeq[TNodePairSeq](t.data.len * GrowthFactor)
```
Comment (Clyybber) [inline compiler/treetab.nim:89]:
```suggestion
      var n = newSeq[TNodePair](t.data.len * GrowthFactor)
```

File: compiler/treetab.nim [hunk 4]
```diff
@@ -103,12 +102,11 @@ proc nodeTableTestOrSet*(t: var TNodeTable, key: PNode, val: int): int =
     result = t.data[index].val
   else:
     if mustRehash(t.data.len, t.counter):
-      var n: TNodePairSeq
-      newSeq(n, t.data.len * GrowthFactor)
+      var n = newSeq[TNodePairSeq](t.data.len * GrowthFactor)
```
Comment (Clyybber) [inline compiler/treetab.nim:105]:
```suggestion
      var n = newSeq[TNodePair](t.data.len * GrowthFactor)
```

**Pull request merged**